src/Common/Core/Impl/OS/Win32EnvironmentBlock.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+using static Microsoft.Common.Core.NativeMethods;
+
+namespace Microsoft.Common.Core.OS {
+    public class Win32EnvironmentBlock : Dictionary<string, string> {
```

Comment:
This should definitely implement `IDictionary` rather than deriving from it, since a bunch of stuff is shadowed with `new` rather than properly overridden. So if you upcast to `Dictionary`, you lose all that special semantics.